### PR TITLE
 GTEST/UCP: client/server API with connection handle testing

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
Testing an API added in #2840 and implemented in #2889

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7899/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov is the test failure 
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7899/label=r-vmb-ppc-jenkins,worker=3/console
```
12:54:04 [ RUN      ] rc/test_ucp_sockaddr_with_rma_atomic.wireup_for_rma_atomic/2
12:54:04 [     INFO ] Testing 65.65.65.12:36080
12:54:05 /scrap/jenkins/workspace/hpc-ucx-pr/label/r-vmb-ppc-jenkins/worker/3/contrib/../test/gtest/common/test.cc:260: Failure
12:54:05 Failed
12:54:05 Got 1 errors during the test
```
related to this PR?

**Comment by evgeny-leksikov** [comment]
partially, should be fixed now

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7904/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7907/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 1]
```diff
@@ -221,13 +241,37 @@ class test_ucp_sockaddr : public ucp_test {
             ucp_request_free(send_req);
         }
 
+        if (send_status == UCS_ERR_REJECTED) {
+            return;
```
Comment (brminich) [inline test/gtest/ucp/test_ucp_sockaddr.cc]: why is it ok?

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 2]
```diff
@@ -241,9 +285,36 @@ class test_ucp_sockaddr : public ucp_test {
...
+               (ucs_get_time() < time_limit)) {
+            check_events(sender().worker(), receiver().worker(), wakeup, NULL);
+        }
+        EXPECT_GT(time_limit, ucs_get_time());
+        EXPECT_EQ(1ul, receiver().get_rejected_cntr());
+    }
+
+    void wait_for_client_reject(bool wakeup)
+    {
+        ucs_time_t time_limit = ucs_get_time() +
```
[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_sockaddr.cc]: fixed

File: test/gtest/ucp/test_ucp_sockaddr.cc (hunk 1 above)
Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_sockaddr.cc]: extended if statement to be clear when it's ok

**Comment by yosefe** [comment]
```
13:48:52 [----------] Global test environment tear-down
13:48:52 [==========] 2795 tests from 598 test cases ran. (1591176 ms total)
13:48:52 [  PASSED  ] 2794 tests.
13:48:52 [  FAILED  ] 1 test, listed below:
13:48:52 [  FAILED  ] rc/test_ucp_sockaddr_with_rma_atomic.wireup_for_rma_atomic/2, where GetParam() = rc
13:48:52 
13:48:52  1 FAILED TEST
13:48:52 
make: *** [test] Error 1
13:48:52 make: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr/label/r-vmb-ppc-jenkins/worker/3/build-test/test/gtest'
13:48:52 
Build step 'Execute shell' mark
```

**Comment by evgeny-leksikov** [comment]
@yosefe log handlers leak somewhere (probably in other tests), will investigate

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7931/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7933/ for details (Mellanox internal link).

**Pull request merged**